File: aidp-python-client/src/aidp_python_client/aidataplatform_dp/models/workspace_object_summary.py
# coding: utf-8
# Copyright (c) 2016, 2026, Oracle and/or its affiliates.  All rights reserved.
# This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.


from oci.util import formatted_flat_dict, NONE_SENTINEL, value_allowed_none_or_none_sentinel  # noqa: F401
from oci.decorators import init_model_state_from_kwargs


@init_model_state_from_kwargs
class WorkspaceObjectSummary(object):
    """
    Summary information about a workspace object.
    """

    def __init__(self, **kwargs):
        """
        Initializes a new WorkspaceObjectSummary object with values from keyword arguments.
        The following keyword arguments are supported (corresponding to the getters/setters of this class):

        :param path:
            The value to assign to the path property of this WorkspaceObjectSummary.
        :type path: str

        :param display_name:
            The value to assign to the display_name property of this WorkspaceObjectSummary.
        :type display_name: str

        :param time_created:
            The value to assign to the time_created property of this WorkspaceObjectSummary.
        :type time_created: datetime

        :param time_updated:
            The value to assign to the time_updated property of this WorkspaceObjectSummary.
        :type time_updated: datetime

        :param metadata:
            The value to assign to the metadata property of this WorkspaceObjectSummary.
        :type metadata: dict(str, str)

        :param composite_etag:
            The value to assign to the composite_etag property of this WorkspaceObjectSummary.
        :type composite_etag: str

        :param type:
            The value to assign to the type property of this WorkspaceObjectSummary.
        :type type: str

        :param description:
            The value to assign to the description property of this WorkspaceObjectSummary.
        :type description: str

        :param system_tags:
            The value to assign to the system_tags property of this WorkspaceObjectSummary.
        :type system_tags: dict(str, dict(str, object))

        :param created_by:
            The value to assign to the created_by property of this WorkspaceObjectSummary.
        :type created_by: str

        :param created_by_name:
            The value to assign to the created_by_name property of this WorkspaceObjectSummary.
        :type created_by_name: str

        """
        self.swagger_types = {
            'path': 'str',
            'display_name': 'str',
            'time_created': 'datetime',
            'time_updated': 'datetime',
            'metadata': 'dict(str, str)',
            'composite_etag': 'str',
            'type': 'str',
            'description': 'str',
            'system_tags': 'dict(str, dict(str, object))',
            'created_by': 'str',
            'created_by_name': 'str'
        }

        self.attribute_map = {
            'path': 'path',
            'display_name': 'displayName',
            'time_created': 'timeCreated',
            'time_updated': 'timeUpdated',
            'metadata': 'metadata',
            'composite_etag': 'compositeEtag',
            'type': 'type',
            'description': 'description',
            'system_tags': 'systemTags',
            'created_by': 'createdBy',
            'created_by_name': 'createdByName'
        }

        self._path = None
        self._display_name = None
        self._time_created = None
        self._time_updated = None
        self._metadata = None
        self._composite_etag = None
        self._type = None
        self._description = None
        self._system_tags = None
        self._created_by = None
        self._created_by_name = None

    @property
    def path(self):
        """
        **[Required]** Gets the path of this WorkspaceObjectSummary.
        The fully qualified path of the workspace object.
        Example: /Shared/Folder1/Notebook1.ipynb


        :return: The path of this WorkspaceObjectSummary.
        :rtype: str
        """
        return self._path

    @path.setter
    def path(self, path):
        """
        Sets the path of this WorkspaceObjectSummary.
        The fully qualified path of the workspace object.
        Example: /Shared/Folder1/Notebook1.ipynb


        :param path: The path of this WorkspaceObjectSummary.
        :type: str
        """
        self._path = path

    @property
    def display_name(self):
        """
        **[Required]** Gets the display_name of this WorkspaceObjectSummary.
        The name of the workspace object. This will be the name of the file/folder in the workspace.
        Example: Notebook1.ipynb, Folder1


        :return: The display_name of this WorkspaceObjectSummary.
        :rtype: str
        """
        return self._display_name

    @display_name.setter
    def display_name(self, display_name):
        """
        Sets the display_name of this WorkspaceObjectSummary.
        The name of the workspace object. This will be the name of the file/folder in the workspace.
        Example: Notebook1.ipynb, Folder1


        :param display_name: The display_name of this WorkspaceObjectSummary.
        :type: str
        """
        self._display_name = display_name

    @property
    def time_created(self):
        """
        **[Required]** Gets the time_created of this WorkspaceObjectSummary.
        The date and time the workspace object was created, in the format defined by <a href=\"https://tools.ietf.org/html/rfc3339\" target=\"_blank\" rel=\"noopener noreferrer\">RFC 3339</a>.
        Example: `2016-08-25T21:10:29.600Z`


        :return: The time_created of this WorkspaceObjectSummary.
        :rtype: datetime
        """
        return self._time_created

    @time_created.setter
    def time_created(self, time_created):
        """
        Sets the time_created of this WorkspaceObjectSummary.
        The date and time the workspace object was created, in the format defined by <a href=\"https://tools.ietf.org/html/rfc3339\" target=\"_blank\" rel=\"noopener noreferrer\">RFC 3339</a>.
        Example: `2016-08-25T21:10:29.600Z`


        :param time_created: The time_created of this WorkspaceObjectSummary.
        :type: datetime
        """
        self._time_created = time_created

    @property
    def time_updated(self):
        """
        Gets the time_updated of this WorkspaceObjectSummary.
        The date and time the workspace object was created, in the format defined by <a href=\"https://tools.ietf.org/html/rfc3339\" target=\"_blank\" rel=\"noopener noreferrer\">RFC 3339</a>.
        Example: `2016-08-25T21:10:29.600Z`


        :return: The time_updated of this WorkspaceObjectSummary.
        :rtype: datetime
        """
        return self._time_updated

    @time_updated.setter
    def time_updated(self, time_updated):
        """
        Sets the time_updated of this WorkspaceObjectSummary.
        The date and time the workspace object was created, in the format defined by <a href=\"https://tools.ietf.org/html/rfc3339\" target=\"_blank\" rel=\"noopener noreferrer\">RFC 3339</a>.
        Example: `2016-08-25T21:10:29.600Z`


        :param time_updated: The time_updated of this WorkspaceObjectSummary.
        :type: datetime
        """
        self._time_updated = time_updated

    @property
    def metadata(self):
        """
        Gets the metadata of this WorkspaceObjectSummary.
        Metadata details of file or folder objects.


        :return: The metadata of this WorkspaceObjectSummary.
        :rtype: dict(str, str)
        """
        return self._metadata

    @metadata.setter
    def metadata(self, metadata):
        """
        Sets the metadata of this WorkspaceObjectSummary.
        Metadata details of file or folder objects.


        :param metadata: The metadata of this WorkspaceObjectSummary.
        :type: dict(str, str)
        """
        self._metadata = metadata

    @property
    def composite_etag(self):
        """
        Gets the composite_etag of this WorkspaceObjectSummary.
        Etag combining data and metadata.


        :return: The composite_etag of this WorkspaceObjectSummary.
        :rtype: str
        """
        return self._composite_etag

    @composite_etag.setter
    def composite_etag(self, composite_etag):
        """
        Sets the composite_etag of this WorkspaceObjectSummary.
        Etag combining data and metadata.


        :param composite_etag: The composite_etag of this WorkspaceObjectSummary.
        :type: str
        """
        self._composite_etag = composite_etag

    @property
    def type(self):
        """
        **[Required]** Gets the type of this WorkspaceObjectSummary.
        The type of workspace object.


        :return: The type of this WorkspaceObjectSummary.
        :rtype: str
        """
        return self._type

    @type.setter
    def type(self, type):
        """
        Sets the type of this WorkspaceObjectSummary.
        The type of workspace object.


        :param type: The type of this WorkspaceObjectSummary.
        :type: str
        """
        self._type = type

    @property
    def description(self):
        """
        Gets the description of this WorkspaceObjectSummary.
        The description of workspace object.


        :return: The description of this WorkspaceObjectSummary.
        :rtype: str
        """
        return self._description

    @description.setter
    def description(self, description):
        """
        Sets the description of this WorkspaceObjectSummary.
        The description of workspace object.


        :param description: The description of this WorkspaceObjectSummary.
        :type: str
        """
        self._description = description

    @property
    def system_tags(self):
        """
        Gets the system_tags of this WorkspaceObjectSummary.
        System tags for this resource. Each key is predefined and scoped to a namespace.

        Example: `{\"orcl-cloud\": {\"free-tier-retained\": \"true\"}}`


        :return: The system_tags of this WorkspaceObjectSummary.
        :rtype: dict(str, dict(str, object))
        """
        return self._system_tags

    @system_tags.setter
    def system_tags(self, system_tags):
        """
        Sets the system_tags of this WorkspaceObjectSummary.
        System tags for this resource. Each key is predefined and scoped to a namespace.

        Example: `{\"orcl-cloud\": {\"free-tier-retained\": \"true\"}}`


        :param system_tags: The system_tags of this WorkspaceObjectSummary.
        :type: dict(str, dict(str, object))
        """
        self._system_tags = system_tags

    @property
    def created_by(self):
        """
        Gets the created_by of this WorkspaceObjectSummary.
        OCID of the user who created this object.


        :return: The created_by of this WorkspaceObjectSummary.
        :rtype: str
        """
        return self._created_by

    @created_by.setter
    def created_by(self, created_by):
        """
        Sets the created_by of this WorkspaceObjectSummary.
        OCID of the user who created this object.


        :param created_by: The created_by of this WorkspaceObjectSummary.
        :type: str
        """
        self._created_by = created_by

    @property
    def created_by_name(self):
        """
        Gets the created_by_name of this WorkspaceObjectSummary.
        Name of the user who created this object.


        :return: The created_by_name of this WorkspaceObjectSummary.
        :rtype: str
        """
        return self._created_by_name

    @created_by_name.setter
    def created_by_name(self, created_by_name):
        """
        Sets the created_by_name of this WorkspaceObjectSummary.
        Name of the user who created this object.


        :param created_by_name: The created_by_name of this WorkspaceObjectSummary.
        :type: str
        """
        self._created_by_name = created_by_name

    def __repr__(self):
        return formatted_flat_dict(self)

    def __eq__(self, other):
        if other is None:
            return False

        return self.__dict__ == other.__dict__

    def __ne__(self, other):
        return not self == other

File: aidp-java-client/src/main/java/com/oracle/aidataplatform/dp/model/WorkspaceObjectSummary.java
// Copyright (c) 2026, Oracle and/or its affiliates.  All rights reserved.

package com.oracle.aidataplatform.dp.model;



/**
 * Summary information about a workspace object.
**/
@jakarta.annotation.Generated(value = "OracleSDKGenerator", comments = "API Version: 20260430")
@com.fasterxml.jackson.databind.annotation.JsonDeserialize(builder=WorkspaceObjectSummary.Builder.class)

public final class WorkspaceObjectSummary  {
    @Deprecated
    @java.beans.ConstructorProperties({"path", "displayName", "timeCreated", "timeUpdated", "metadata", "compositeEtag", "type", "description", "systemTags", "createdBy", "createdByName"})
    public WorkspaceObjectSummary(String path, String displayName, java.util.Date timeCreated, java.util.Date timeUpdated, java.util.Map<String, String> metadata, String compositeEtag, WorkspaceObjectDetails.Type type, String description, java.util.Map<String, java.util.Map<String, Object>> systemTags, String createdBy, String createdByName) {
        super();
        this.path = path;
        this.displayName = displayName;
        this.timeCreated = timeCreated;
        this.timeUpdated = timeUpdated;
        this.metadata = metadata;
        this.compositeEtag = compositeEtag;
        this.type = type;
        this.description = description;
        this.systemTags = systemTags;
        this.createdBy = createdBy;
        this.createdByName = createdByName;
    }

    @com.fasterxml.jackson.databind.annotation.JsonPOJOBuilder(withPrefix = "")
    public static class Builder {
                /**
     * The fully qualified path of the workspace object.
* Example: /Shared/Folder1/Notebook1.ipynb
* 
     **/
    
@com.fasterxml.jackson.annotation.JsonProperty("path")
private String path;

        /**
         * The fully qualified path of the workspace object.
* Example: /Shared/Folder1/Notebook1.ipynb
* 
         * @param path the value to set
         * @return this builder
         **/
        

public Builder path(String path) {
    this.path = path;
    return this;
}
            /**
     * The name of the workspace object. This will be the name of the file/folder in the workspace.
* Example: Notebook1.ipynb, Folder1
* 
     **/
    
@com.fasterxml.jackson.annotation.JsonProperty("displayName")
private String displayName;

        /**
         * The name of the workspace object. This will be the name of the file/folder in the workspace.
* Example: Notebook1.ipynb, Folder1
* 
         * @param displayName the value to set
         * @return this builder
         **/
        

public Builder displayName(String displayName) {
    this.displayName = displayName;
    return this;
}
            /**
     * The date and time the workspace object was created, in the format defined by <a href="https://tools.ietf.org/html/rfc3339" target="_blank" rel="noopener noreferrer">RFC 3339</a>.
* Example: {@code 2016-08-25T21:10:29.600Z}
* 
     **/
    
@com.fasterxml.jackson.annotation.JsonProperty("timeCreated")
private java.util.Date timeCreated;

        /**
         * The date and time the workspace object was created, in the format defined by <a href="https://tools.ietf.org/html/rfc3339" target="_blank" rel="noopener noreferrer">RFC 3339</a>.
* Example: {@code 2016-08-25T21:10:29.600Z}
* 
         * @param timeCreated the value to set
         * @return this builder
         **/
        

public Builder timeCreated(java.util.Date timeCreated) {
    this.timeCreated = timeCreated;
    return this;
}
            /**
     * The date and time the workspace object was created, in the format defined by <a href="https://tools.ietf.org/html/rfc3339" target="_blank" rel="noopener noreferrer">RFC 3339</a>.
* Example: {@code 2016-08-25T21:10:29.600Z}
* 
     **/
    
@com.fasterxml.jackson.annotation.JsonProperty("timeUpdated")
private java.util.Date timeUpdated;

        /**
         * The date and time the workspace object was created, in the format defined by <a href="https://tools.ietf.org/html/rfc3339" target="_blank" rel="noopener noreferrer">RFC 3339</a>.
* Example: {@code 2016-08-25T21:10:29.600Z}
* 
         * @param timeUpdated the value to set
         * @return this builder
         **/
        

public Builder timeUpdated(java.util.Date timeUpdated) {
    this.timeUpdated = timeUpdated;
    return this;
}
            /**
     * Metadata details of file or folder objects.
     **/
    
@com.fasterxml.jackson.annotation.JsonProperty("metadata")
private java.util.Map<String, String> metadata;

        /**
         * Metadata details of file or folder objects.
         * @param metadata the value to set
         * @return this builder
         **/
        

public Builder metadata(java.util.Map<String, String> metadata) {
    this.metadata = metadata;
    return this;
}
            /**
     * Etag combining data and metadata.
     **/
    
@com.fasterxml.jackson.annotation.JsonProperty("compositeEtag")
private String compositeEtag;

        /**
         * Etag combining data and metadata.
         * @param compositeEtag the value to set
         * @return this builder
         **/
        

public Builder compositeEtag(String compositeEtag) {
    this.compositeEtag = compositeEtag;
    return this;
}
            /**
     * The type of workspace object.
     **/
    
@com.fasterxml.jackson.annotation.JsonProperty("type")
private WorkspaceObjectDetails.Type type;

        /**
         * The type of workspace object.
         * @param type the value to set
         * @return this builder
         **/
        

public Builder type(WorkspaceObjectDetails.Type type) {
    this.type = type;
    return this;
}
            /**
     * The description of workspace object.
     **/
    
@com.fasterxml.jackson.annotation.JsonProperty("description")
private String description;

        /**
         * The description of workspace object.
         * @param description the value to set
         * @return this builder
         **/
        

public Builder description(String description) {
    this.description = description;
    return this;
}
            /**
     * System tags for this resource. Each key is predefined and scoped to a namespace.
* <p>
Example: {@code {"orcl-cloud": {"free-tier-retained": "true"}}}
* 
     **/
    
@com.fasterxml.jackson.annotation.JsonProperty("systemTags")
private java.util.Map<String, java.util.Map<String, Object>> systemTags;

        /**
         * System tags for this resource. Each key is predefined and scoped to a namespace.
* <p>
Example: {@code {"orcl-cloud": {"free-tier-retained": "true"}}}
* 
         * @param systemTags the value to set
         * @return this builder
         **/
        

public Builder systemTags(java.util.Map<String, java.util.Map<String, Object>> systemTags) {
    this.systemTags = systemTags;
    return this;
}
            /**
     * OCID of the user who created this object.
     **/
    
@com.fasterxml.jackson.annotation.JsonProperty("createdBy")
private String createdBy;

        /**
         * OCID of the user who created this object.
         * @param createdBy the value to set
         * @return this builder
         **/
        

public Builder createdBy(String createdBy) {
    this.createdBy = createdBy;
    return this;
}
            /**
     * Name of the user who created this object.
     **/
    
@com.fasterxml.jackson.annotation.JsonProperty("createdByName")
private String createdByName;

        /**
         * Name of the user who created this object.
         * @param createdByName the value to set
         * @return this builder
         **/
        

public Builder createdByName(String createdByName) {
    this.createdByName = createdByName;
    return this;
}


        public WorkspaceObjectSummary build() {
            WorkspaceObjectSummary model = new WorkspaceObjectSummary(this.path
                , this.displayName
                , this.timeCreated
                , this.timeUpdated
                , this.metadata
                , this.compositeEtag
                , this.type
                , this.description
                , this.systemTags
                , this.createdBy
                , this.createdByName);            return model;
        }

        @com.fasterxml.jackson.annotation.JsonIgnore
        public Builder copy(WorkspaceObjectSummary model) {
                this.path(model.getPath());
    this.displayName(model.getDisplayName());
    this.timeCreated(model.getTimeCreated());
    this.timeUpdated(model.getTimeUpdated());
    this.metadata(model.getMetadata());
    this.compositeEtag(model.getCompositeEtag());
    this.type(model.getType());
    this.description(model.getDescription());
    this.systemTags(model.getSystemTags());
    this.createdBy(model.getCreatedBy());
    this.createdByName(model.getCreatedByName());
return this;
        }
    }

    /**
     * Create a new builder.
     */
    public static Builder builder() {
        return new Builder();
    }


    public Builder toBuilder() {
        return new Builder().copy(this);
    }

    


        /**
     * The fully qualified path of the workspace object.
* Example: /Shared/Folder1/Notebook1.ipynb
* 
     **/
    
    @com.fasterxml.jackson.annotation.JsonProperty("path")
    private final String path;

        /**
     * The fully qualified path of the workspace object.
* Example: /Shared/Folder1/Notebook1.ipynb
* 
     * @return the value
     **/
    
    public String getPath() {
        return path;
    }


        /**
     * The name of the workspace object. This will be the name of the file/folder in the workspace.
* Example: Notebook1.ipynb, Folder1
* 
     **/
    
    @com.fasterxml.jackson.annotation.JsonProperty("displayName")
    private final String displayName;

        /**
     * The name of the workspace object. This will be the name of the file/folder in the workspace.
* Example: Notebook1.ipynb, Folder1
* 
     * @return the value
     **/
    
    public String getDisplayName() {
        return displayName;
    }


        /**
     * The date and time the workspace object was created, in the format defined by <a href="https://tools.ietf.org/html/rfc3339" target="_blank" rel="noopener noreferrer">RFC 3339</a>.
* Example: {@code 2016-08-25T21:10:29.600Z}
* 
     **/
    
    @com.fasterxml.jackson.annotation.JsonProperty("timeCreated")
    private final java.util.Date timeCreated;

        /**
     * The date and time the workspace object was created, in the format defined by <a href="https://tools.ietf.org/html/rfc3339" target="_blank" rel="noopener noreferrer">RFC 3339</a>.
* Example: {@code 2016-08-25T21:10:29.600Z}
* 
     * @return the value
     **/
    
    public java.util.Date getTimeCreated() {
        return timeCreated;
    }


        /**
     * The date and time the workspace object was created, in the format defined by <a href="https://tools.ietf.org/html/rfc3339" target="_blank" rel="noopener noreferrer">RFC 3339</a>.
* Example: {@code 2016-08-25T21:10:29.600Z}
* 
     **/
    
    @com.fasterxml.jackson.annotation.JsonProperty("timeUpdated")
    private final java.util.Date timeUpdated;

        /**
     * The date and time the workspace object was created, in the format defined by <a href="https://tools.ietf.org/html/rfc3339" target="_blank" rel="noopener noreferrer">RFC 3339</a>.
* Example: {@code 2016-08-25T21:10:29.600Z}
* 
     * @return the value
     **/
    
    public java.util.Date getTimeUpdated() {
        return timeUpdated;
    }


        /**
     * Metadata details of file or folder objects.
     **/
    
    @com.fasterxml.jackson.annotation.JsonProperty("metadata")
    private final java.util.Map<String, String> metadata;

        /**
     * Metadata details of file or folder objects.
     * @return the value
     **/
    
    public java.util.Map<String, String> getMetadata() {
        return metadata;
    }


        /**
     * Etag combining data and metadata.
     **/
    
    @com.fasterxml.jackson.annotation.JsonProperty("compositeEtag")
    private final String compositeEtag;

        /**
     * Etag combining data and metadata.
     * @return the value
     **/
    
    public String getCompositeEtag() {
        return compositeEtag;
    }


        /**
     * The type of workspace object.
     **/
    
    @com.fasterxml.jackson.annotation.JsonProperty("type")
    private final WorkspaceObjectDetails.Type type;

        /**
     * The type of workspace object.
     * @return the value
     **/
    
    public WorkspaceObjectDetails.Type getType() {
        return type;
    }


        /**
     * The description of workspace object.
     **/
    
    @com.fasterxml.jackson.annotation.JsonProperty("description")
    private final String description;

        /**
     * The description of workspace object.
     * @return the value
     **/
    
    public String getDescription() {
        return description;
    }


        /**
     * System tags for this resource. Each key is predefined and scoped to a namespace.
* <p>
Example: {@code {"orcl-cloud": {"free-tier-retained": "true"}}}
* 
     **/
    
    @com.fasterxml.jackson.annotation.JsonProperty("systemTags")
    private final java.util.Map<String, java.util.Map<String, Object>> systemTags;

        /**
     * System tags for this resource. Each key is predefined and scoped to a namespace.
* <p>
Example: {@code {"orcl-cloud": {"free-tier-retained": "true"}}}
* 
     * @return the value
     **/
    
    public java.util.Map<String, java.util.Map<String, Object>> getSystemTags() {
        return systemTags;
    }


        /**
     * OCID of the user who created this object.
     **/
    
    @com.fasterxml.jackson.annotation.JsonProperty("createdBy")
    private final String createdBy;

        /**
     * OCID of the user who created this object.
     * @return the value
     **/
    
    public String getCreatedBy() {
        return createdBy;
    }


        /**
     * Name of the user who created this object.
     **/
    
    @com.fasterxml.jackson.annotation.JsonProperty("createdByName")
    private final String createdByName;

        /**
     * Name of the user who created this object.
     * @return the value
     **/
    
    public String getCreatedByName() {
        return createdByName;
    }

    @Override
    public String toString() {
        return this.toString(true);
    }

    /**
     * Return a string representation of the object.
     * @param includeByteArrayContents true to include the full contents of byte arrays
     * @return string representation
     */
    public String toString(boolean includeByteArrayContents) {
        java.lang.StringBuilder sb = new java.lang.StringBuilder();
        sb.append("WorkspaceObjectSummary(");
        sb.append("path=").append(String.valueOf(this.path));
        sb.append(", displayName=").append(String.valueOf(this.displayName));
        sb.append(", timeCreated=").append(String.valueOf(this.timeCreated));
        sb.append(", timeUpdated=").append(String.valueOf(this.timeUpdated));
        sb.append(", metadata=").append(String.valueOf(this.metadata));
        sb.append(", compositeEtag=").append(String.valueOf(this.compositeEtag));
        sb.append(", type=").append(String.valueOf(this.type));
        sb.append(", description=").append(String.valueOf(this.description));
        sb.append(", systemTags=").append(String.valueOf(this.systemTags));
        sb.append(", createdBy=").append(String.valueOf(this.createdBy));
        sb.append(", createdByName=").append(String.valueOf(this.createdByName));
        sb.append(")");
        return sb.toString();
    }

    @Override
    public boolean equals(Object o) {
        if (this == o) {
            return true;
        }
        if (!(o instanceof WorkspaceObjectSummary)) {
            return false;
        }

        WorkspaceObjectSummary other = (WorkspaceObjectSummary) o;
        return java.util.Objects.equals(this.path, other.path) &&
            java.util.Objects.equals(this.displayName, other.displayName) &&
            java.util.Objects.equals(this.timeCreated, other.timeCreated) &&
            java.util.Objects.equals(this.timeUpdated, other.timeUpdated) &&
            java.util.Objects.equals(this.metadata, other.metadata) &&
            java.util.Objects.equals(this.compositeEtag, other.compositeEtag) &&
            java.util.Objects.equals(this.type, other.type) &&
            java.util.Objects.equals(this.description, other.description) &&
            java.util.Objects.equals(this.systemTags, other.systemTags) &&
            java.util.Objects.equals(this.createdBy, other.createdBy) &&
            java.util.Objects.equals(this.createdByName, other.createdByName);
    }

    @Override
    public int hashCode() {
        final int PRIME = 59;
        int result = 1;
        result = (result * PRIME) + (this.path == null ? 43 : this.path.hashCode());
        result = (result * PRIME) + (this.displayName == null ? 43 : this.displayName.hashCode());
        result = (result * PRIME) + (this.timeCreated == null ? 43 : this.timeCreated.hashCode());
        result = (result * PRIME) + (this.timeUpdated == null ? 43 : this.timeUpdated.hashCode());
        result = (result * PRIME) + (this.metadata == null ? 43 : this.metadata.hashCode());
        result = (result * PRIME) + (this.compositeEtag == null ? 43 : this.compositeEtag.hashCode());
        result = (result * PRIME) + (this.type == null ? 43 : this.type.hashCode());
        result = (result * PRIME) + (this.description == null ? 43 : this.description.hashCode());
        result = (result * PRIME) + (this.systemTags == null ? 43 : this.systemTags.hashCode());
        result = (result * PRIME) + (this.createdBy == null ? 43 : this.createdBy.hashCode());
        result = (result * PRIME) + (this.createdByName == null ? 43 : this.createdByName.hashCode());
        return result;
    }


}
